versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/implrocksdb/RocksDBStoreManager.java

```diff
@@ -0,0 +1,278 @@
+
+    try (final Options options = new Options()) {
+      options.setCreateIfMissing(true);
+      // get a list of existing families.
+      families = new ArrayList<>(RocksDB.listColumnFamilies(options, baseDirectory));
+    } catch (RocksDBException e) {
+      LOGGER.error("Error extracting RocksDB options", e);
+    }
+
+    // if empty, add the default family
+    if (families != null && families.isEmpty()) {
+      families.addAll(COLUMN_FAMILIES);
+    }
+    final Function<byte[], ColumnFamilyDescriptor> func = ColumnFamilyDescriptor::new;
+
+    List<ColumnFamilyHandle> familyHandles = new ArrayList<>();
+    try (final DBOptions dboptions = new DBOptions()) {
+      dboptions.setCreateIfMissing(true);
+
+      // TODO what options need to be set?
+
+      // registerMetrics(dboptions);
+      try {
+        db = openDB(dboptions, baseDirectory, new ArrayList<>(Lists.transform(families, func)), familyHandles);
+
+      } catch (RocksDBException e) {
+        LOGGER.error("Error opening RocksDB backing store", e);
+      }
+    }
+    // populate the local cache with the existing tables.
+    for (int i = 0; i < families.size(); i++) {
+      byte[] family = families.get(i);
+      if (Arrays.equals(family, RocksDB.DEFAULT_COLUMN_FAMILY)) {
+        defaultHandle = familyHandles.get(i);
+      } else {
+        String name = new String(family, UTF_8);
+        final ColumnFamilyHandle handle = familyHandles.get(i);
+        handleIdToNameMap.put(handle.getID(), name);
+        RocksDBStore store = newRocksDBStore(name, new ColumnFamilyDescriptor(family), handle);
+        maps.put(name, store);
+      }
+    }
+  }
+
+  /**
+   * Opens or creates a Rocks DB.
+   *
+   * @param dboptions     the options with which to open the database
+   * @param path          where the database is located
+   * @param columnNames   an array of columns in the database
+   * @param familyHandles the handles of the column families.
+   * @return the instance of RocksDB created or opened
+   * @throws RocksDBException Exception with RocksDB
+   */
+  public RocksDB openDB(final DBOptions dboptions, final String path, final List<ColumnFamilyDescriptor> columnNames,
+                        List<ColumnFamilyHandle> familyHandles) throws RocksDBException {
+    boolean printLockMessage = true;
+
+    while (true) {
+      try {
+        return RocksDB.open(dboptions, path, columnNames, familyHandles);
+      } catch (RocksDBException e) {
+        if (e.getStatus().getCode() != Status.Code.IOError || !e.getStatus().getState().contains("While lock")) {
+          throw e;
+        }
+
+        // Don't retry if request came from a CLI command (noDBOpenRetry = true)
+        if (noDBOpenRetry) {
+          LOGGER.error("Lock file to RocksDB is currently held by another process.  Stop other process before retrying.");
+          System.out.println("Lock file to RocksDB is currently held by another process.  Stop other process before retrying.");
```

Comment:
This seems problematic to log to sout....is that how it happens in the rest of the codebase?

Reply:
Oops. No, this came from the sample code. I'll use LOGGER.